src/libraries/System.Private.CoreLib/src/System/IO/TextWriter.cs

```diff
@@ -125,6 +125,17 @@ public virtual void Write(char value)
         {
         }
 
+        public virtual void Write(Rune value)
+        {
+            // Convert value to span
+            Span<char> chars = stackalloc char[2];
+            int charsWritten = value.EncodeToUtf16(chars);
+            ReadOnlySpan<char> charsSlice = chars[..charsWritten];
+
+            // Write span
+            Write(charsSlice);
```

Comment:
This is going to rent array from the ArrayPool and then call the method taking array. I don't think we need to do that here.This can be implemented as 

```csharp
            Span<char> chars = stackalloc char[2];
            int charsWritten = value.EncodeToUtf16(chars);
            Write(chars[0]);

            if (charsWritten > 1)
            {
                Write(chars[1]);
            }
```